.github/JAVASCRIPT_CODESTYLE.md

```diff
@@ -0,0 +1,563 @@
+
+```js
+var a = [
+  'hello', 'world'
+];
+var b = {"good": 'code'
+        , is generally: 'pretty'
+        };
+```
+
+## Conditionals
+
+### Use the === operator
+
+Programming is not about remembering [stupid rules][comparisonoperators]. Use
+the triple equality operator as it will work just as expected.
+
+*Right:*
+
+```js
+var a = 0;
+if (a !== '') {
+  console.log('winning');
+}
+
+```
+
+*Wrong:*
+
+```js
+var a = 0;
+if (a == '') {
+  console.log('losing');
+}
+```
+
+[comparisonoperators]: https://developer.mozilla.org/en/JavaScript/Reference/Operators/Comparison_Operators
+
+### Use multi-line ternary operator
+
+The ternary operator should not be used on a single line. Split it up into multiple lines instead.
+
+*Right:*
+
+```js
+var foo = (a === b)
+  ? 1
+  : 2;
+```
+
+*Wrong:*
+
+```js
+var foo = (a === b) ? 1 : 2;
+```
+
+### Use descriptive conditions
+
+Any non-trivial conditions should be assigned to a descriptively named variable or function:
+
+*Right:*
+
+```js
+var isValidPassword = password.length >= 4 && /^(?=.*\d).{4,}$/.test(password);
+
+if (isValidPassword) {
+  console.log('winning');
+}
+```
+
+*Wrong:*
+
+```js
+if (password.length >= 4 && /^(?=.*\d).{4,}$/.test(password)) {
+  console.log('losing');
+}
+```
+
+## Functions
+
+### Write small functions
+
+Keep your functions short. A good function fits on a slide that the people in
+the last row of a big room can comfortably read. So don't count on them having
+perfect vision and limit yourself to ~15 lines of code per function.
+
+### Return early from functions
+
+To avoid deep nesting of if-statements, always return a function's value as early
+as possible.
+
+*Right:*
+
+```js
+function isPercentage(val) {
+  if (val < 0) {
+    return false;
+  }
+
+  if (val > 100) {
+    return false;
+  }
+
+  return true;
+}
+```
+
+*Wrong:*
+
+```js
+function isPercentage(val) {
+  if (val >= 0) {
+    if (val < 100) {
+      return true;
+    } else {
+      return false;
+    }
+  } else {
+    return false;
+  }
+}
+```
+
+Or for this particular example it may also be fine to shorten things even
+further:
+
+```js
+function isPercentage(val) {
+  var isInRange = (val >= 0 && val <= 100);
+  return isInRange;
+}
+```
+
+### Name your closures
+
+Feel free to give your closures a name. It shows that you care about them, and
+will produce better stack traces, heap and cpu profiles.
+
+*Right:*
+
+```js
+req.on('end', function onEnd() {
+  console.log('winning');
+});
+```
+
+*Wrong:*
+
+```js
+req.on('end', function() {
+  console.log('losing');
+});
+```
+
+### No nested closures
+
+Use closures, but don't nest them. Otherwise your code will become a mess.
+
+*Right:*
+
+```js
+setTimeout(function() {
+  client.connect(afterConnect);
+}, 1000);
+
+function afterConnect() {
+  console.log('winning');
+}
+```
+
+*Wrong:*
+
+```js
+setTimeout(function() {
+  client.connect(function() {
+    console.log('losing');
+  });
+}, 1000);
+```
+
+
+### Method chaining
```

Comment:
I also am a bit weary about this. It would make large modifications on other files necessary.